### PR TITLE
♻️ Refactors projects plugin sub-modules as controller-service-repository

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Reorganizing webserver's projects plugin to reflect [controller-service-repository pattern](https://tom-collings.medium.com/controller-service-repository-16e29a4684e5). 
image

For instance, let's assume a project's sub-resource `foo`. It should be split in three submodules as
 - `projects/_foo_handlers.py` : controler
 - `projects/_foo_api.py`: service 
 - `projects/_foo_db.py`: repository

This PR adapts only **some** of the submodules, namely:
- `projects/ports`
- `projects/permalinks`
- `projects/nodes`
- `projects/tags`
- **deprecated** nodes submodule was removed. i.e. they are not used.

## Related issue/s

- follows up pattern above as proposed by @matusdrobuliak66 
- follows up on `projects` extensions of https://github.com/ITISFoundation/osparc-simcore/pull/4383 and https://github.com/ITISFoundation/osparc-simcore/pull/4379

## How to test

In-place

## DevOps

None

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]
great! thanks :+1:

**Review by GitHK — COMMENTED** [review]
A few considerations:
- I would personally prefer to use `*_repository.py` in place of `*_db.py` since: it can be Postregres, it can be Redis, it can be S3.
- why are the nodes api endpoints deleted? I found nothing in the description of the PR

**Comment by pcrespov** [comment]
> * I would personally prefer to use `*_repository.py` in place of `*_db.py` since: it can be Postregres, it can be Redis, it can be S3.

- IMO `db` is good enough, otherwise the names become far too long. Same way that we do not write `service` but `api`

> * why are the nodes api endpoints deleted? I found nothing in the description of the PR

- They were **deprecated** (it was actually written in the description but now i have highlighted it). We never used them.

File: services/web/server/src/simcore_service_webserver/projects/_nodes_api.py [hunk 1]
```diff
@@ -1,24 +1,71 @@
...
 from .._constants import APP_SETTINGS_KEY, RQT_USERID_KEY
+from ..application_settings import get_settings
 from ..storage.api import get_download_link
+from .exceptions import ProjectStartsTooManyDynamicNodes
 
 _logger = logging.getLogger(__name__)
+_NODE_START_INTERVAL_S: Final[datetime.timedelta] = datetime.timedelta(seconds=15)
+
+
+def get_service_start_lock_key(user_id: UserID, project_uuid: ProjectID) -> str:
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/projects/_nodes_api.py:33]: hmm... very much looking to find that refactoring in my PR...

**Review by sanderegg — APPROVED** [review]
ok. if possible, wait for my PR about node resources to go in before pushing this one.

File: services/web/server/src/simcore_service_webserver/projects/_nodes_api.py (hunk 1 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/projects/_nodes_api.py:33]: @sanderegg I wait for your PR to get in

**Pull request merged**